Source/Diagnostics/ComputeDiagFunctors/ParticleReductionFunctor.H

```diff
@@ -26,12 +26,14 @@ public:
      * \param[in] fn_str parser string that describes the function to apply to particles
      *            and then average over each cell
      * \param[in] ispec index of the species over which to calculate the average
+     * \param[in] do_average Whether to do an average or a sum of the function
      * \param[in] do_filter Whether to apply a filter function to particles before averaging
      * \param[in] filter_str Parser string for filter function to apply before averaging
      * \param[in] ncomp Number of component of mf_src to cell-center in dst multifab.
      */
     ParticleReductionFunctor(const amrex::MultiFab * const mf_src, const int lev,
-                       const amrex::IntVect crse_ratio, const std::string fn_str, const int ispec,
+                       const amrex::IntVect crse_ratio, const std::string fn_str,
+                       const int ispec, const int do_average,
```

Comment:
Is there a particular reason for `do_average` to be an int rather than a bool? If not, it might make sense for it to be a bool in order to parallel `do_filter` (or for `do_filter` to also be an int).

Reply:
I tend to avoid `vector<bool>` since that type can be problematic. It would work ok here though since only the `[]` operator is used. I can make the change.

Reply:
Gotcha. All things being equal, I think my personal inclination is towards bool here, especially since the `std::vector<bool>` should be okay, but if there's a convention against it, I'm fine with keeping it as an int. I know there's been some discussion about int vs bool at recent dev meetings, but I forget whether it was just re: avoiding `amrex::Vector<bool>` in device code.

Reply:
Sorry, I might have missed something here. It seems like now we do have `vector<bool>` in Source/Diagnostics/Diagnostics.H (the member variable `m_pfield_do_average`), so why would it be problematic to replace the argument `const int do_average` with `const bool do_average` here (and in the other instances below)?
```suggestion
                       const int ispec, const bool do_average,
```

Reply:
@EZoni Your changes are correct - it was an oversight that the `do_average` scalar was not changed to a bool.